tests/libcaf/test_repository_tags.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from pathlib import Path
+from libcaf.repository import Repository, RepositoryError, HashRef
+from libcaf.constants import HASH_LENGTH
+from libcaf.tag import TagNotInTagsDirError, TagAlreadyExistsError
+
+@pytest.fixture
+def commit_hash(temp_repo: Repository) -> HashRef:
+    """Fixture to create a single commit and return its hash."""
+    (temp_repo.working_dir / 'file.txt').write_text('content')
+    return temp_repo.commit_working_dir('Test Author', 'Test Message')
+
+def test_init_creates_tags_dir(temp_repo: Repository):
+    """Tests that the 'tags' directory is created by repo.init()."""
+    assert temp_repo.tags_dir().exists()
+    assert temp_repo.tags_dir().is_dir()
+
+def test_object_exists(temp_repo: Repository, commit_hash: HashRef):
+    """Tests the object_exists helper for valid and invalid hashes."""
+    assert temp_repo.object_exists(commit_hash)
+    assert not temp_repo.object_exists('a' * HASH_LENGTH)
+    assert not temp_repo.object_exists('invalid-hash')
+    assert not temp_repo.object_exists('')
+
+def test_create_tag_success(temp_repo: Repository, commit_hash: HashRef):
+    """Tests the happy path for creating a tag."""
+    temp_repo.create_tag('v1.0', commit_hash)
+    tag_path = temp_repo.tags_dir() / 'v1.0'
```

Comment:
Again testing implementation details, which is easy to break if some of the details change